File: backend/entity/food_flag.go
package entity

import (
	"gorm.io/gorm"
)

type FoodFlag struct {
	gorm.Model
	Flag string

	// FoodGroup ทำหน้าที่เป็น FK
	FoodGroupID uint
	FoodGroup FoodGroup `gorm:"foreignKey:FoodGroupID"`

	// 1 FoodFlag เป็นเจ้าของได้หลาย FoodItem
	FoodItems []FoodItem `gorm:"foreignKey:FoodFlagID"`

}

File: frontend/src/interfaces/FoodFlag.ts
import type { FoodGroupInterface } from "./FoodGroup";

export interface FoodFlagInterface {
    ID?: number;
    Flag?: string;
    FoodGroupID?: number; //FK          
}

export type FoodFlagType = {
  ID: number;
  Flag: string;
  FoodGroup: FoodGroupInterface;
};